app/src/main/java/com/github/swent/echo/compose/map/LocationSelector.kt

```diff
@@ -0,0 +1,56 @@
+package com.github.swent.echo.compose.map
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import com.github.swent.echo.R
+import com.mapbox.mapboxsdk.geometry.LatLng
+import org.ramani.compose.CameraPosition
+import org.ramani.compose.Circle
+import org.ramani.compose.MapLibre
+import org.ramani.compose.Symbol
+
+@Composable
+fun LocationSelector(
+    modifier: Modifier = Modifier,
+    initialLocation: LatLng = MAP_CENTER.toLatLng(),
+    locationCallback: (LatLng) -> Unit
+) {
+    val context = LocalContext.current
+    val styleUrl =
+        context.getString(R.string.maptiler_base_style_url) +
+            context.getString(R.string.maptiler_api_key)
+    var markerCenter by remember { mutableStateOf(initialLocation) }
+    var mapCenter by remember { mutableStateOf(initialLocation) }
+    MapLibre(
+        modifier = modifier.testTag("location-selector-map-libre"),
+        styleUrl = styleUrl,
+        cameraPosition = CameraPosition(target = mapCenter, zoom = DEFAULT_ZOOM)
+    ) {
+        Circle(
+            center = markerCenter,
+            radius = 30F,
+            isDraggable = true,
+            color = "Red",
+            opacity = 0.1F,
+            zIndex = 1,
+            onCenterDragged = { markerCenter = it },
+            onDragFinished = {
+                mapCenter = it
```

Comment:
Making the map center every time one dragged the marker also seems to make it zoom in on that point. This is not really useful since one most likely has to zoom out again to actually get closer to the exact location one wanted to select. Disabling the centering here would make the location selector easier to use.